views/blog/_archive.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+/**
+ * @var \Yiisoft\Data\Reader\DataReaderInterface|string[][] $archive
+ * @var \Yiisoft\Router\UrlGeneratorInterface $urlGenerator
+ * @var \Yiisoft\View\WebView $this
+ */
+
+use Yiisoft\Html\Html;
+
+?>
+<h4 class="text-muted mb-3">
+    Archive
+</h4>
+<ul class="list-group mb-3">
+    <?php
+    $currentYear = null;
+    $blockBegin = Html::beginTag(
+        'li',
+        ['class' => 'list-group-item d-flex flex-column justify-content-between lh-condensed']
+    );
+    $blockEnd = Html::endTag('li');
+    if (count($archive)) {
+        foreach ($archive->read() as $aValue) {
```

Comment:
It's better to use longer but descriptive names. For `$aValue` it could be `$item` or `$post`.